packaging/resource_suite_s3_nocache.py

```diff
@@ -2406,6 +2409,67 @@ def test_upload_very_large_file_after_redirect_issues_2104_2114(self):
             s3plugin_lib.remove_if_exists(file1)
             s3plugin_lib.remove_if_exists(retrieved_file)
 
+    def test_iput_with_N_flag__issue_2198(self):
+
+        filename = f'{inspect.currentframe().f_code.co_name}'
+        get_filename = f'{filename}.get'
+        try:
+            file_sizes = [
+                    100,
+                    32 * 1024 * 1024 - 1,
+                    32 * 1024 * 1024,
+                    32 * 1024 * 1024 + 1,
+                    100 * 1024 * 1024
+                    ];
+
+            for file_size in file_sizes:
+                # create the file
+                lib.make_arbitrary_file(filename, file_size)
+
+                # put the file
+                self.user1.assert_icommand(f'iput {filename}')
+                self.user1.assert_icommand(f'iget -f {filename} {get_filename}')
+                assert_command(['diff', '-q', filename, get_filename])
+
+                for n_value in [0, 1, 2,3,4,8]:
+                    self.user1.assert_icommand(f'iput -f -N{n_value} {filename}')
+                    self.user1.assert_icommand(f'iget -f {filename} {get_filename}')
+                    assert_command(['diff', '-q', filename, get_filename])
+
+                # cleanup before next loop
+                s3plugin_lib.remove_if_exists(filename)
+                s3plugin_lib.remove_if_exists(get_filename)
+                self.user1.assert_icommand(f'irm -f {filename}')
+
+        finally:
+            # clean up the file that was put in case it wasn't cleaned up
+            self.user1.assert_icommand(f'irm -f {filename}')
+
+    def recursive_put_small_files_with_checksum_and_N_flag__issue_2198(self):
+
+        dirname = f'{inspect.currentframe().f_code.co_name}'
+        filename1 = f'{inspect.currentframe().f_code.co_name}1'
+        filename2 = f'{inspect.currentframe().f_code.co_name}1'
+        filename3 = f'{inspect.currentframe().f_code.co_name}3'
+        try:
+            # create the local directory
+            os.mkdir(dirname)
+            # create the files
+            lib.make_arbitrary_file(f'{dirname}/{filename1}', 100)
+            lib.make_arbitrary_file(f'{dirname}/{filename2}', 100)
+            lib.make_arbitrary_file(f'{dirname}/{filename3}', 100)
+
+            # recursively put the directory
+            self.user1.assert_icommand(f'iput -r -N2 -K {dirname}')
+
+            #verify the files are in iRODS 
```

Comment:
Should we also verify the contents in some way? Don't necessarily need to verify checksums, but maybe an `iget` with a `diff` like the other test as well as maybe checking for the existence of a checksum in the catalog could be enough... Not sure if that's overkill for what this is testing. If so, maybe an explanatory comment or more words in the test name could explain that?

Reply:
done